pandas/core/arrays/base.py

```diff
@@ -400,6 +400,36 @@ def dropna(self):
 
         return self[~self.isna()]
 
+    def shift(self, periods=1):
```

Comment:
I think update in the ExtensionArray doc-string?

Reply:
Currently, in the class docstring we only mention the methods that either needs to implemented (because they raise AbstractMethodError otherwise) or either have a suboptimal implementation because it does the object ndarray roundtrip. 
This is not the case here (which is not saying we couldn't also list other methods that can be overriden for specific reasons)